### DISCUSSION TITLE
Remote debugging from VS Code

### DISCUSSION THREAD

**Discussion by NoaTzur** [microsoft/vscode-discussions#1113] [Extension Development QnA]
Hey, 

I'm using vs code on ubuntu machine. 
I want to connect to a remote device that runs gdbserver.

I'm running this command in the remote device 
`gdbserver :2345 my-program`

and this is the prints i get after :

```
Process my-program created; pid = 1518
Listening on port 2345
```

Now in the vs code, i have created a lanuch.json file with this content : 

```
{
    // Use IntelliSense to learn about possible attributes.
    // Hover to view descriptions of existing attributes.
    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",
    "configurations": [
        {
        "type": "gdb",
        "request": "attach",
        "name": "Attach to gdbserver",
        "executable": "./my-program",
        "gdbpath": "/usr/bin/gdb-multiarch",
        "target": "192.168.30.151:2345",
        "remote": true,
        "cwd": "${workspaceRoot}",
        }
    ]
}
```

* at the remote device I'm running the same binary "my-program" that I compiled in the vscode and moved with scp to the remote device.
 
after pressing F5, I can see the `Remote debugging from host 172.17.137.106` message.
I can press "continue", "step over" and even put a breakpoints.

but - after pressing "continue" and the program is running, I can no longer control it through the vs code.
breakpoint does not work, when pressing "pause" the program in the remote device continue running ...

netstat on the remote shows that my machine with the vs code still connected : 

`tcp        0      0 192.168.30.151:2345     172.17.137.106:35284    ESTABLISHED
`
can anyone help me ? am I missing some important configuration ?

Thanks !

**Comment by isidorn** [comment]
This community is for vscode extension authors. I am closing this since this seems like an issue with using VS Code.

**Reply by NoaTzur** [reply]
Sorry, can you please direct me to the right forum where I can ask this question? 

Thanks!

**Reply by isidorn** [reply]
For questions, I suggest stackoverflow.
If you have a bug with reproduible steps then you can file it here https://github.com/microsoft/vscode/issues
